app/waf/src/controllers/wafpolicy.controller.ts

```diff
@@ -120,6 +129,104 @@ export class WafpolicyController extends BaseController {
     );
   }
 
+  @post(prefix + '/wafpolicies/{id}/adcs/{adcId}', {
+    responses: {
+      '204': Schema.emptyResponse('Uploading WAF Policy resource'),
+      '404': Schema.notFound('Can not find trustedDeivceId'),
+    },
+  })
+  async wafpolicyUpload(
+    @param(Schema.pathParameter('id', 'WAF Policy resource ID')) id: string,
+    @param(Schema.pathParameter('adcId', 'ADC resource ID')) adcId: string,
+  ): Promise<void> {
+    let wafpolicy: Wafpolicy = await this.wafpolicyRepository.findById(id, {
+      where: {
+        and: [
+          {or: [{tenantId: await this.tenantId}, {public: true}]},
+          {id: id},
+        ],
+      },
+    });
+
+    if (wafpolicy === undefined) {
```

Comment:
I remember findById() will throw error, if not found. Am I wrong?

Reply:
Yes it throws entity not found error